# Olypics Data Analysis
## CS 176 Project

By: Srushti Vaidyanathan (vaidyan4@purdue.edu), Paige Burke (burke231@purdue.edu), Brooklyn Fugate (fugate11@purdue.edu)

## Set Up

In [102]:
# importing pandas as pd, matplotlib as plt, and numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# read datasets from their csv files to the dataframe
athletes_df = pd.read_csv('olympic_atheletes_history_biometrics_data.csv')
summer_df = pd.read_csv('summer_olympics_medals_data.csv')
winter_df = pd.read_csv('winter_olympics_medals_data.csv')

In [71]:
athletes_df.head(5)

ID              ATHLETE_NAME SEX   AGE  HEIGHT  WEIGHT  NOC         TEAM  \
0   1                 A Dijiang   M  24.0   180.0    80.0  CHN        China   
1   2                  A Lamusi   M  23.0   170.0    60.0  CHN        China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  DEN      Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  DEN      Denmark   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0  NED  Netherlands   

   Unnamed: 8  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN

In [72]:
summer_df.head(5)

Year                       Name        Games    City      Sport  \
0  1896     Aristidis Akratopoulos  1896 Summer  Athina     Tennis   
1  1896     Aristidis Akratopoulos  1896 Summer  Athina     Tennis   
2  1896  Konstantinos Akratopoulos  1896 Summer  Athina     Tennis   
3  1896  Konstantinos Akratopoulos  1896 Summer  Athina     Tennis   
4  1896         Anastasios Andreou  1896 Summer  Athina  Athletics   

                                Event Medal  
0                Tennis Men's Singles   NaN  
1                Tennis Men's Doubles   NaN  
2                Tennis Men's Singles   NaN  
3                Tennis Men's Doubles   NaN  
4  Athletics Men's 110 metres Hurdles   NaN

In [73]:
winter_df.head(5)

Year                 Name        Games      City                 Sport  \
0  1924   Clarence John Abel  1924 Winter  Chamonix            Ice Hockey   
1  1924          Josef Adolf  1924 Winter  Chamonix       Nordic Combined   
2  1924  Xavier Affentranger  1924 Winter  Chamonix           Ski Jumping   
3  1924  Xavier Affentranger  1924 Winter  Chamonix  Cross Country Skiing   
4  1924  Xavier Affentranger  1924 Winter  Chamonix       Nordic Combined   

                                       Event   Medal  
0                Ice Hockey Men's Ice Hockey  Silver  
1           Nordic Combined Men's Individual     NaN  
2  Ski Jumping Men's Normal Hill, Individual     NaN  
3   Cross Country Skiing Men's 18 kilometres     NaN  
4           Nordic Combined Men's Individual     NaN

## Cleaning

CLEANING IS ALSO DONE THROUGHOUT THE OTHER STEPS

In [110]:
# deleting this column as it was empty and was just an extra comma in the csv
athletes_df = athletes_df.drop('Unnamed: 8', axis=1)
athletes_df = athletes_df.drop('ID', axis=1)

In [132]:
# standardizing to date time format for all the dfs
winter_df['Year'] = pd.to_datetime(winter_df['Year'], format='%Y')
summer_df['Year'] = pd.to_datetime(summer_df['Year'], format='%Y')

winter_df['Year'] = winter_df['Year'].dt.strftime('%Y')
summer_df['Year'] = summer_df['Year'].dt.strftime('%Y')

Year                    Name        Games      City  \
0      1924      Clarence John Abel  1924 Winter  Chamonix   
1      1924             Josef Adolf  1924 Winter  Chamonix   
2      1924     Xavier Affentranger  1924 Winter  Chamonix   
3      1924     Xavier Affentranger  1924 Winter  Chamonix   
4      1924     Xavier Affentranger  1924 Winter  Chamonix   
...     ...                     ...          ...       ...   
48559  2014  Stepan Olegovich Zuyev  2014 Winter     Sochi   
48560  2014      Kristaps Zvejnieks  2014 Winter     Sochi   
48561  2014      Kristaps Zvejnieks  2014 Winter     Sochi   
48562  2014                Piotr ya  2014 Winter     Sochi   
48563  2014                Piotr ya  2014 Winter     Sochi   

                      Sport                                      Event   Medal  
0                Ice Hockey                Ice Hockey Men's Ice Hockey  Silver  
1           Nordic Combined           Nordic Combined Men's Individual     NaN  
2               Ski Jumping  Ski Jumping Men's Normal Hill, Individual     NaN  
3      Cross Country Skiing   Cross Country Skiing Men's 18 kilometres     NaN  
4           Nordic Combined           Nordic Combined Men's Individual     NaN  
...                     ...                                        ...     ...  
48559         Alpine Skiing                 Alpine Skiing Men's Slalom     NaN  
48560         Alpine Skiing           Alpine Skiing Men's Giant Slalom     NaN  
48561         Alpine Skiing                 Alpine Skiing Men's Slalom     NaN  
48562           Ski Jumping   Ski Jumping Men's Large Hill, Individual     NaN  
48563           Ski Jumping         Ski Jumping Men's Large Hill, Team     NaN  

[48561 rows x 7 columns]

In [112]:

# drops any duplicate values from the dfs and only keeps the first value
winter_df = winter_df.drop_duplicates(keep='first')
summer_df = summer_df.drop_duplicates(keep='first')
athletes_df = athletes_df.drop_duplicates(keep='first')

Markdown notes and sentences and analysis written here.

## Filtering

In [114]:
# Creating a subset of biometrics of just males and females, respectively, to compare using visualizations

female_athletes_df = athletes_df[athletes_df['SEX'] == 'F']
male_athletes_df = athletes_df[athletes_df['SEX'] == 'M']

Markdown notes and sentences and analysis written here.

## Sorting

In [ ]:
# code here

Markdown notes and sentences and analysis written here.

## Merging

In [161]:
# merging the winter_df with summer_df using inner to only have the rows of athletes who have participated in both the summer and winter olympics

overlapping_athletes_df = pd.merge(winter_df, summer_df, on=['Name'], how='inner', suffixes=['_w','_s'])

# filter the data to only include rows where the Summer and Winter participation years were within 20 years of one another
overlapping_athletes_df = overlapping_athletes_df[abs(overlapping_athletes_df['Year_w'].astype(int) - overlapping_athletes_df['Year_s'].astype(int)) <= 20]

print(len(overlapping_athletes_df.Name.unique()))
display(overlapping_athletes_df.tail())

275


Year_w                    Name      Games_w City_w    Sport_w  \
1739   2014  Lauryn Chenet Williams  2014 Winter  Sochi  Bobsleigh   
1740   2014  Lauryn Chenet Williams  2014 Winter  Sochi  Bobsleigh   
1741   2014  Lauryn Chenet Williams  2014 Winter  Sochi  Bobsleigh   
1742   2014  Lauryn Chenet Williams  2014 Winter  Sochi  Bobsleigh   
1743   2014  Lauryn Chenet Williams  2014 Winter  Sochi  Bobsleigh   

                    Event_w Medal_w Year_s      Games_s   City_s    Sport_s  \
1739  Bobsleigh Women's Two  Silver   2004  2004 Summer   Athina  Athletics   
1740  Bobsleigh Women's Two  Silver   2004  2004 Summer   Athina  Athletics   
1741  Bobsleigh Women's Two  Silver   2008  2008 Summer  Beijing  Athletics   
1742  Bobsleigh Women's Two  Silver   2008  2008 Summer  Beijing  Athletics   
1743  Bobsleigh Women's Two  Silver   2012  2012 Summer   London  Athletics   

                                     Event_s Medal_s  
1739            Athletics Women's 100 metres  Silver  
1740  Athletics Women's 4 x 100 metres Relay     NaN  
1741            Athletics Women's 100 metres     NaN  
1742  Athletics Women's 4 x 100 metres Relay     NaN  
1743  Athletics Women's 4 x 100 metres Relay    Gold

In regards to the merge above:

The overlapping_athletes_df above is very interesting in the fact that it covers athletes that have competed in both the Summer and the Winter Olympics. It starts by merging the winter_df and the summer_df on the 'Name' column, using the 'inner' merge technique. This allows us to only retain the rows with athletes who have a name under the summer_df and winter_df. However, upon initially analyzing this data, we noticed that some of the athletes had a lengthy gap between their Summer and Winter record (ex. a Summer record 40-60 years apart from their Winter record). This made us realize that there are several athletes with the same name. For this reason, to make the data more accurate, we filtered the data to include only the rows with the Summer year and Winter year value to be within 20 years of each other, as an average athlete won't attend more than 5 Olympic cycles in their lifetime. This filtered the data to be more accurate, such as Lauryn Williams, shown above, who participated in both bobsleigh and athletics. Some of the data however, may still be inaccurate, as several people have had the same names, as well as the fact that for several years, some of the Winter Sports were held along side the Summer Sports. However, generally, this merged and filtered data set well represents the data of athletes who have participated in both the Winter and Summer Olympics. There are 275 unique athletes in this data frame who have participated in both seasons.

In [120]:
# merging the winter_df with summer_df using outer to have all of the rows of both data sets
all_seasons_df = pd.merge(winter_df, summer_df, on=['Year', 'Name', 'Games', 'City', 'Sport', 'Event', 'Medal'], how='outer')

all_seasons_df.head(5)

Year                 Name        Games      City                 Sport  \
0  1924   Clarence John Abel  1924 Winter  Chamonix            Ice Hockey   
1  1924          Josef Adolf  1924 Winter  Chamonix       Nordic Combined   
2  1924  Xavier Affentranger  1924 Winter  Chamonix           Ski Jumping   
3  1924  Xavier Affentranger  1924 Winter  Chamonix  Cross Country Skiing   
4  1924  Xavier Affentranger  1924 Winter  Chamonix       Nordic Combined   

                                       Event   Medal  
0                Ice Hockey Men's Ice Hockey  Silver  
1           Nordic Combined Men's Individual     NaN  
2  Ski Jumping Men's Normal Hill, Individual     NaN  
3   Cross Country Skiing Men's 18 kilometres     NaN  
4           Nordic Combined Men's Individual     NaN

In regards to the merge above:

The new dataframe that was created, called 'all_seasons_df' combines the winter_df and summer_df to have a new dataframe containing data of all instances of an athlete's performance in the Olympics. It will contain the data of every event and the winner from both the summer and the winter olympics. We combined these data frames using the common columns: Year, Name, Games, City, Sport, Event, and Medal.

The main use for this data frame will be to merge this with the athletes_df to get a combined dataframe of all of the data, further explained below.

In [116]:
# Merging all three data sets into a total data set

# first merges the athletes df with winter df
total_df = pd.merge(athletes_df, all_seasons_df, left_on='ATHLETE_NAME', right_on='Name')
total_df = total_df.drop('Name', axis=1)

In regards to the merge above:

The new dataframe that was created, called 'total_df' contains a combined dataframe with the rows and columns from all three of the original data sets, winter_df, summer_df, and athletes_df. It was merged using the all_seasons_df (explained above), and the athletes_df using the name of the athlete. 

We will use this data frame first to clean the data in the athletes_df, by stacking using the gender and sport and sorting these indices, to impute data for the age, height and weight columns of the dataframe. The reason we sort by the gender and sport is so that we can impute similar biometrics for the athletes with missing data (ex. not accidentally using a male wrestlers' weight for a female figure skater).
Next, 

In [117]:
# Cleaning height, weight, and age of total_df by stack and sorting
# by gender and sport, then imputing, to get similar data to fill in the NA values

total_df = total_df.set_index(['SEX', 'Sport']).sort_index()
total_df['AGE'] = total_df['AGE'].ffill()
total_df['HEIGHT'] = total_df['HEIGHT'].ffill()
total_df['WEIGHT'] = total_df['WEIGHT'].ffill()

Markdown notes and sentences and analysis written here.

## Visualizations

In [ ]:
# code here

Markdown notes and sentences and analysis written here.

## Pivoting and Stacking

In [ ]:
# code here

Markdown notes and sentences and analysis written here.